pkg/interface/src/views/components/OverlaySigil.tsx

```diff
@@ -139,3 +133,5 @@ export default class OverlaySigil extends PureComponent<OverlaySigilProps, Overl
     );
   }
 }
+
+export default withLocalState(OverlaySigil);
```

Comment:
It'd be nice if this was a seperate component that only passed down the props it needed

Reply:
Do you have an example of this pattern?

Reply:
It's not strictly necessary for this, given the infrequency with which the localState tends to change, but if we're serious about hookifying the whole state, then we could end up with a boatload of spurious rerenders. (Bridge currently has this problem in spades). Here's a nicer version of `withLocalState` that would cause less rerenders. Dunno if these types work, but you could definitely strongly type such a function. 
```typescript
function withLocalState<P, S extends keyof LocalState>(pickState: S[], Component: (compProps: P) => ReactNode) {
  return React.forwardRef((props: Omit<P,S>, ref) => {
    const localState = _.pick(useLocalState(), pickState);
    return <Component ref={ref} {...localState} {...props} />
  })
}
function withLocalState<P>(component: Component, relea

Reply:
Ok, nice. I see what you mean. Zustand actually has a nice way of selectively using state members. I'll get this written up tonight.